quint/src/ir/IRVisitor.ts

```diff
@@ -261,14 +265,23 @@ export function walkDeclaration(visitor: IRVisitor, decl: ir.QuintDeclaration):
     visitor.enterDecl(decl)
   }
 
+  // The standard depth starts at 0, so definitions inside delclarations (i.e.
+  // assume and instance overrides) are not considered top-level
+  visitor.definitionDepth = 0
+
   switch (decl.kind) {
     case 'const':
     case 'var':
-    case 'def':
     case 'typedef':
     case 'assume':
       walkDefinition(visitor, decl)
       break
+    case 'def':
+      // depth will be increased inside `walkDefinition`, so we set it to -1 in
+      // order for it to be 0 there
+      visitor.definitionDepth = -1
+      walkDefinition(visitor, decl)
```

Comment:
I think I am a bit lost in the difference between top-level and inner definitions. Would not that code get triggered on both?

Reply:
Top-level definitions are declarations, while inner definitions are NOT declarations. This is inside `walkDeclaration`, so the code would only be called right before walking top level definitions.